tests/Tasks/UpdatePackageInfoTest.php

```diff
@@ -46,30 +57,59 @@ public function testGetPackageInfo()
 
     public function testPackagesAreAddedCorrectly()
     {
-        $task = UpdatePackageInfoTask::create();
+        // Get the latest supported framework version from the MetaData service
+        $data = MetaData::getMetaDataForRepository('silverstripe/silverstripe-framework')['majorVersionMapping'];
+        $keys = array_keys($data);
+        $key = $keys[count($keys) - 1];
+        $latestFrameworkMajor = $data[$key][0];
 
-        $frameworkVersion = VersionProvider::singleton()->getModuleVersion('silverstripe/framework');
-        $composerLoader = $this->getMockBuilder(ComposerLoader::class)
-            ->setMethods(['getLock'])->getMock();
-        $composerLoader->expects($this->any())->method('getLock')->will($this->returnValue(json_decode(<<<LOCK
-{
-    "packages": [
-        {
-            "name": "silverstripe/framework",
-            "description": "A faux package from a mocked composer.lock for testing purposes",
-            "version": "$frameworkVersion"
-        },
-        {
-            "name": "fake/unsupported-package",
-            "description": "A faux package from a mocked composer.lock for testing purposes",
-            "version": "1.0.0"
-        }
-    ],
-    "packages-dev": null
-}
-LOCK
-        )));
-        $task->setComposerLoader($composerLoader);
+        // Mock the VersionProvider service to return the latest framework version
+        Injector::inst()->registerService(new class($latestFrameworkMajor) extends VersionProvider {
+            private $latestFrameworkMajor;
+            public function __construct($latestFrameworkMajor)
+            {
+                $this->latestFrameworkMajor = $latestFrameworkMajor;
+            }
+            public function getModuleVersion(string $module): string
+            {
+                if ($module === 'silverstripe/framework') {
+                    return $this->latestFrameworkMajor . '.0.0';
+                }
+                return parent::getVersion($module);
+            }
+        }, VersionProvider::class);
+
+        // Create a task and mock the ComposerLoader to return a specific composer.lock
+        $task = UpdatePackageInfoTask::create();
```

Comment:
Refactoring of $this->getMockBuilder() to use an anoymous class so that it matches VersionProvider mocking above